### PR TITLE
Fix #964 (printing)

### LINKED ISSUE #964: no method matching mapcontainer_warn(::JuMP.#_getValue, ::JuMP.JuMPArray{JuMP.NonlinearExpression,1,Tuple{UnitRange{Int64}}})
```julia
using JuMP
M = [0  0 -1 -1 ;
     0  0  1 -2 ;
     1 -1  2 -2 ;
     1  2 -2  4 ]

q = [2; 2; -2; -6]

lb = zeros(4)
ub = Inf*ones(4)

items = 1:4
m = Model()
@variable(m, lb[i] <= x[i in items] <= ub[i])
@NLexpression(m, F[i in items], sum(M[i,j]*x[j] for j in items) + q[i])
@objective(m, Min, 1)
solve(m)

getvalue(x) # this works
getvalue(F[1]) # this works
getvalue(F) # this does NOT work
```

produces an error:

```julia
ERROR: MethodError: no method matching mapcontainer_warn(::JuMP.#_getValue, ::JuMP.JuMPArray{JuMP.NonlinearExpression,1,Tuple{UnitRange{Int64}}})
Closest candidates are:
  mapcontainer_warn(::Any, ::JuMP.JuMPContainer{JuMP.Variable,N}) at /Users/chkwon/.julia/v0.5/JuMP/src/JuMP.jl:824
 in _map(::Function, ::JuMP.JuMPArray{JuMP.NonlinearExpression,1,Tuple{UnitRange{Int64}}}) at /Users/chkwon/.julia/v0.5/JuMP/src/JuMPContainer.jl:150
 in getvalue(::JuMP.JuMPArray{JuMP.NonlinearExpression,1,Tuple{UnitRange{Int64}}}) at /Users/chkwon/.julia/v0.5/JuMP/src/JuMPContainer.jl:110
```

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

**Comment by blegat** [comment]
The failure of AppVeyor is unrelated to this PR, I have pushed a fixed for it in #975

**Comment by blegat** [comment]
I have rebased it again master, appveyor and travis should work now, let's see...

**Comment by mlubin** [comment]
CC @joehuchette for review

File: test/print.jl [hunk 1]
```diff
@@ -218,8 +222,7 @@ end
       [10,Banana,:]
         [10,Banana,-1] = -10.0
         [10,Banana, 0] = 0.0
-        [10,Banana, 1] = 10.0
-    """, repl=:print)
+        [10,Banana, 1] = 10.0""", repl=:print)
```
Comment (joehuchette) [inline test/print.jl]: Why change this behavior?

File: src/JuMPContainer.jl [hunk 2]
```diff
@@ -99,6 +99,7 @@ end
 
 pushmeta!(x::JuMPContainer, sym::Symbol, val) = (x.meta[sym] = val)
 getmeta(x::JuMPContainer, sym::Symbol) = x.meta[sym]
+hasmeta(x::JuMPContainer, sym::Symbol) = haskey(x.meta, sym)
```
Comment (joehuchette) [inline src/JuMPContainer.jl]: Would be good to write this in such a way that it won't fail obscurely for user-defined types descended from ``JuMPContainer`` that don't have a ``meta`` field. Maybe something like
```jl
metadata(x::Union{JuMPArray,JuMPDict}) = x.meta
metadata(::JuMPContainer) = error("Useful error message here")
pushmeta!(x::JuMPContainer, sym::Symbol) = (metadata(x)[sym] = val)
getmeta(x::JuMPContainer, sym::Symbol) = metadata(x)[sym]
hasmeta(x::JuMPContainer, sym::Symbol) = haskey(metadata(x), sym)
```

Comment (mlubin) [inline src/JuMPContainer.jl]: Yeah, assuming fields on abstract types isn't great.

Comment (joehuchette) [inline src/JuMPContainer.jl]: This is good with me to merge now, but this should be addressed separately then

File: test/print.jl (hunk 1 above)
Comment (blegat) [inline test/print.jl]: This was inconsistent. Some terminated by `\n` like `JuMPArray{Float64}`, some did not like `JuMPDict{Float64}`. I figured that it was more standard not to terminate with `\n`, e.g. `Array` does not terminate with `\n` so I changed `JuMPArray{Float64}` to that :)

Comment (joehuchette) [inline test/print.jl]: mmk, sounds good

File: src/JuMPContainer.jl (hunk 2 above)
Comment (blegat) [inline src/JuMPContainer.jl]: I have addressed this in [this branch](https://github.com/blegat/JuMP.jl/tree/meta). Once this PR is merged, I will rebase it and do a PR

Comment (joehuchette) [inline src/JuMPContainer.jl]: Great, thanks (I didn't mean to imply you should do that work, just that it shouldn't be forgotten)

Comment (blegat) [inline src/JuMPContainer.jl]: Don't worry, I just wanted to ;)

**Pull request merged**